### ISSUE TITLE
dependency-track is taking 25% of CPU usage with no ptojects

### ISSUE THREAD

**Issue by somera** [DependencyTrack/dependency-track#3236]
### Current Behavior

I run dependecy-tack for the first time.

```
docker run -d \
    --network=host \
    -m 8192m -p 7070:8080 \
    --name dependency-track \
    -v /data-files/data/docker/dependency-track:/data \
    dependencytrack/bundled
```

The initial process was successful. After 2-3 hourd I didn't see no changes in the log. But dependecy-tack is using 25% of CPU usage on my NUC.  It's empty. Without any projects. Why so much CPU usage?

At 14:17 I restarted dependency-track.

![image](https://github.com/DependencyTrack/dependency-track/assets/8334250/97dd5b15-6465-4034-8130-413b4e3f0525)

![image](https://github.com/DependencyTrack/dependency-track/assets/8334250/7d06d95d-6b87-4318-908e-5803d67d723d)

Without dependency-track the CPU usage is ~2-3%.

### Steps to Reproduce

Just start the Docker Container.

### Expected Behavior

Less CPU usage when no projects was added.

### Dependency-Track Version

4.9.1

### Dependency-Track Distribution

Container Image

### Database Server

H2

### Browser

Google Chrome

### Checklist

- [X] I have read and understand the [contributing guidelines](https://github.com/DependencyTrack/dependency-track/blob/master/CONTRIBUTING.md#filing-issues)
- [X] I have checked the [existing issues](https://github.com/DependencyTrack/dependency-track/issues) for whether this defect was already reported

**Comment by nscuro** [comment]
Indeed DT should not use as much CPU when idle. I just launched an empty 4.9.1 instance on my workstation, attached VisualVM, and will let it run like this for a few hours.  I suspect that H2's internal maintenance plays a factor here, but we'll see.

**Comment by somera** [comment]
I startet my setup on other NUC and will take a look on this.

**Comment by nscuro** [comment]
Yeah so even after letting it run for ~15min, I can already tell it's the H2 database's maintenance thread that keeps working. Notice the *Total Time (CPU)* numbers:

![image](https://github.com/DependencyTrack/dependency-track/assets/5693141/2850451e-398d-40b3-9cb3-7e5d6fb2f0c9)

With `docker stats` I'm seeing similar numbers to what you are reporting:

```
CONTAINER ID   NAME                           CPU %     MEM USAGE / LIMIT     MEM %     NET I/O          BLOCK I/O       PIDS
47831309e8a6   dependency-track-apiserver-1   35.13%    3.399GiB / 11.67GiB   29.14%    23.9MB / 880MB   48MB / 30.4GB   76
```

And at this point in time, DT itself ran idle.

Generally, H2 is good for initial testing, but for everything beyond that you should use a dedicated DB server (preferably PostgreSQL). They are much more efficient and get the job done with fewer resources.

**Comment by somera** [comment]
Thx for the input. Than I will use my PostgreSQL 16 instance for DT.

If you think there is nothing to do, than this can be closed.

**Comment by nscuro** [comment]
I have launched a PostgreSQL setup with the same monitoring connected to it. Just want to *really* confirm that the CPU usage indeed settles in that case. I'll close this issue once that is confirmed.

**Comment by somera** [comment]
Only short question.

I have working PostgreSQL 16 with

`password_encryption = scram-sha-256`

Is DT support this?

**Comment by nscuro** [comment]
AFAIK there's no special treatment needed on the client side to work with a DB configured with `password_encryption = scram-sha-256`. Just provide your database credentials as usual and it should just work.

**Comment by somera** [comment]
It works. All I had to do was add a suitable line in `pg_hba.conf`.
